### PR TITLE
Add Notification about new FeedbackRequest

### PR DESCRIPTION
**Pull request by Farleo**
add status to feedback request
delete duration column
daily check available student and feedback request

### CODE REVIEW

**Review by Arsen-Kushnir — APPROVED** [review]

File: src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java [hunk 1]
```diff
@@ -32,12 +32,10 @@ public static User fromDto(UserDto dto) {
...
 	
 	public static List<UserDto> toDto(List<User> users) {
-		UserDtoMapper userDtoMapper = new UserDtoMapper();
 		return Objects.isNull(users) ? null : users.stream().map(UserDtoMapper::toDto).collect(Collectors.toList());
 	}
 	
 	public static List<User> fromDto(List<UserDto> entities) {
-		UserDtoMapper userDtoMapper = new UserDtoMapper();
 		return Objects.isNull(entities) ? null : entities.stream().map(UserDtoMapper::fromDto).collect(Collectors.toList());
 	}
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java:40]: if null return empty list

File: src/main/java/ita/softserve/course_evaluation/repository/FeedbackRequestRepository.java [hunk 2]
```diff
@@ -14,6 +14,9 @@
 	
 	@Query(value = "SELECT * From course_feedback_request where course_id = ?1", nativeQuery = true)
 	Page<FeedbackRequest> findAllByCourseId (Pageable page, Long id);
+	
+	@Query(value = "SELECT * From course_feedback_request cfr where cfr.status = ?1 and (CURRENT_DATE >= cast(cfr.start_date as date) and CURRENT_DATE <= cast(cfr.end_date as date))", nativeQuery = true)
+	List<FeedbackRequest> findAllByStatusAndValidDate(Long id);
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/repository/FeedbackRequestRepository.java:20]: use one style for sql commands   (SELECT  From where)

File: src/main/java/ita/softserve/course_evaluation/repository/UserRepository.java [hunk 3]
```diff
@@ -32,16 +32,23 @@
...
+    @Query(value = "SELECT u.* FROM course_group cg " +
+                           "INNER JOIN users u on cg.group_id = u.group_id " +
+                           "INNER JOIN course_feedback_request cfr on cg.course_id = cfr.course_id " +
+                           "LEFT JOIN course_feedback cf on u.id = cf.student_id " +
+                           "WHERE cfr.id = :id " +
+                           "and (cfr.id <> cf.feedback_request_id or cf.id is null)" +
+                           "and (cfr.status=1 or cfr.status=2)" +
+                           "and (CURRENT_DATE >= cast(cfr.start_date as date) and CURRENT_DATE <= cast(cfr.end_date as date))", nativeQuery = true)
+    List<User> findAllUserByFeedbackRequestIdWithoutFeedback(long id);
 }
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/repository/UserRepository.java:54]: use one style for sql commands (SELECT on and)

File: src/main/java/ita/softserve/course_evaluation/service/impl/NotificationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			emailContext.setSubject("Feedback request to " +  courseName + " available");
+			emailContext.put("FeedbackRequestDescription", feedbackRequest.getFeedbackDescription());
+			emailContext.put("CourseName", courseName);
+			emailContext.put("siteUrl", siteUrl);
+			try {
+				emailService.sendMail(emailContext);
+			} catch (MessagingException e) {
+				e.printStackTrace();
+			}
+		}
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/service/impl/NotificationServiceImpl.java:58]: use log

File: src/test/java/ita/softserve/course_evaluation/repository/QuestionRepositoryTest.java [hunk 5]
```diff
@@ -91,7 +94,7 @@ void testFindAllQuestionIdsByFeedbackRequest(){
 
         FeedbackRequest feedbackRequest = new FeedbackRequest(1L, "description",
                 LocalDateTime.now(), LocalDateTime.now().plusDays(10),
-                10L, course, Collections.emptyList(), List.of(question1, question2, question3));
+                course, Collections.emptyList(), List.of(question1, question2, question3), FeedbackRequestStatus.values()[1], LocalDateTime.now());
         feedbackRequestRepository.save(feedbackRequest);
         System.out.println(questionRepository.findAll());
         System.out.println(feedbackRequest);
```
Comment (VovaProtsiv) [inline src/test/java/ita/softserve/course_evaluation/repository/QuestionRepositoryTest.java:100]: remove System.out.println

**Review by VovaProtsiv — CHANGES_REQUESTED** [review]

**Review by VovaProtsiv — APPROVED** [review]

**Pull request merged**